### PR TITLE
Change setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by samuelgarcia**

### CODE REVIEW

**Comment by samuelgarcia** [comment]
Hi @caniko.
Here a first draft for pyproject.toml.
It is not poetry based (sorry) we stick with setuptools for now.
Any feeedback welcome.

Do you know a way to also include other files (matlab source) to be able to remove the old MANIFEST.in ?

**Comment by caniko** [comment]
Great! I'll take a look later today.

Regarding `MANIFEST.in`, with [poetry](https://python-poetry.org/docs/pyproject/#include-and-exclude); `setuptools` has [no support](https://github.com/pypa/setuptools/issues/3341) for this currently.

**Comment by caniko** [comment]
Hope the feedback is constructive! Otherwise, LGTM :100:

**Review by alejoe91 — APPROVED** [review]

**Comment by caniko** [comment]
You didn't see this earlier? @alejoe91 @samuelgarcia

**Comment by alejoe91** [comment]
@caniko nope! Thanks for the constructive feedback! We'll make a separate PR!

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,126 @@
+[project]
+name = "spikeinterface"
+version = "0.95.1.dev0"
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply pyproject.toml:3]: we keep this in between releases. All released versions are MAJOR.MINOR.PATCH

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+    ## sliding_nn
+    "pymde",
+    "torch",
+    "pynndescent",
+    
+    # probeinterface and neo from master
+    "probeinterface @ git+https://github.com/SpikeInterface/probeinterface.git",
+    "neo @ git+https://github.com/NeuralEnsemble/python-neo.git",
```
[the remark this answers is not in the document]
Comment (alejoe91) [reply pyproject.toml:120]: changes in NEO, probeinterface, and SpikeInterface are in sync, so for tests we prefer to test the current version

[the remark this answers is not in the document]
Comment (caniko) [reply pyproject.toml:120]: I understand the practicality, but it makes the testing non-deterministic, shouldn't every package be atomic?

**Pull request merged**